kotlin/navigate-route-with-rerouting/src/main/java/com/esri/arcgisruntime/sample/navigateroutewithrerouting/MainActivity.kt

```diff
@@ -0,0 +1,396 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.navigateroutewithrerouting
+
+import android.graphics.Color
+import android.os.Bundle
+import android.widget.Button
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.ArcGISRuntimeEnvironment
+import com.esri.arcgisruntime.geometry.Geometry
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.geometry.Polyline
+import com.esri.arcgisruntime.geometry.SpatialReferences
+import com.esri.arcgisruntime.location.SimulatedLocationDataSource
+import com.esri.arcgisruntime.location.SimulationParameters
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.BasemapStyle
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.mapping.view.LocationDisplay
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.navigation.DestinationStatus
+import com.esri.arcgisruntime.navigation.ReroutingParameters
+import com.esri.arcgisruntime.navigation.RouteTracker
+import com.esri.arcgisruntime.navigation.TrackingStatus
+import com.esri.arcgisruntime.sample.navigateroutewithrerouting.databinding.ActivityMainBinding
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import com.esri.arcgisruntime.symbology.SimpleMarkerSymbol
+import com.esri.arcgisruntime.tasks.networkanalysis.DirectionManeuver
+import com.esri.arcgisruntime.tasks.networkanalysis.Route
+import com.esri.arcgisruntime.tasks.networkanalysis.RouteParameters
+import com.esri.arcgisruntime.tasks.networkanalysis.RouteResult
+import com.esri.arcgisruntime.tasks.networkanalysis.RouteTask
+import com.esri.arcgisruntime.tasks.networkanalysis.Stop
+import java.util.Calendar
+import kotlin.math.roundToInt
+
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    private val activityMainBinding by lazy {
+        ActivityMainBinding.inflate(layoutInflater)
+    }
+
+    private val mapView: MapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val navigateButton: Button by lazy {
+        activityMainBinding.navigateButton
+    }
+
+    private val resetButton: Button by lazy {
+        activityMainBinding.resetButton
+    }
+
+    private val recenterButton: Button by lazy {
+        activityMainBinding.recenterButton
+    }
+
+    private val statusMessageTV: TextView by lazy {
+        activityMainBinding.statusMessageTV
+    }
+
+    // a list to keep track of directions solved by the route task
+    private val directionManeuvers = mutableListOf<DirectionManeuver>()
```

Comment:
I wonder if it's worth keeping a second list of the maneuvers. Couldn't we just reference route.directionManeuvers directly where we need it?

Reply:
Removed `directionManeuvers` list 👍

Reply:
Turns out we need it haha.